### PR TITLE
Skip tests if no CUDA devices are present

### PR DESCRIPTION
**Pull request by fwyzard**
If no CUDA devices are present, skip the tests and return a success status, instead of failing.

### CODE REVIEW

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by fwyzard** [comment]
type bugfix

**Comment by fwyzard** [comment]
@cmsbuild, please test

File: DataFormats/Math/test/cudaAtan2Test.cu [hunk 1]
```diff
@@ -59,75 +56,76 @@ __global__ void diffAtan(int * diffs) {
...
-   std::cout <<	"a non cuda error" << std::endl;
-}
-
-
-  return 0;
+  int count = 0;
+  auto status = cudaGetDeviceCount(& count);
+  if (status != cudaSuccess) {
+    std::cerr << "Failed to initialise the CUDA runtime, the test will be skipped." << "\n";
+    exit(EXIT_SUCCESS);
```
Comment (makortel) [inline DataFormats/Math/test/cudaAtan2Test.cu]: This is cosmetics, but why `exit` and not `return`?

Comment (fwyzard) [reply DataFormats/Math/test/cudaAtan2Test.cu]: Because I copy-pasted it from the `go` function, and did not change `exit` into `return`.

**Comment by fwyzard** [comment]
All failures seem to be EOS-related:
```
Failed to open the file 'root://cms-xrd-global.cern.ch//store/...
```

**Comment by fwyzard** [comment]
Feel free to re-test and merge at your convenience.

**Comment by smuzaffar** [comment]
please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/24938	16a49e9
- technical; in line with the PR description; affects only cuda test files in DataFormats/Math/test/
- jenkins tests pass

**Comment by fabiocos** [comment]
+1

**Pull request merged**